sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java

```diff
@@ -0,0 +1,145 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.collect;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.blob.v2.BlobIndices;
+import io.crate.core.collections.Row;
+import io.crate.executor.transport.TransportActionProvider;
+import io.crate.integrationtests.SQLHttpIntegrationTest;
+import io.crate.lucene.CrateDocIndexService;
+import io.crate.metadata.Functions;
+import io.crate.metadata.Routing;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.shard.ShardReferenceResolver;
+import io.crate.metadata.shard.blob.BlobShardReferenceResolver;
+import io.crate.operation.reference.doc.blob.BlobReferenceResolver;
+import io.crate.planner.distribution.DistributionInfo;
+import io.crate.planner.node.dql.RoutedCollectPhase;
+import io.crate.planner.projection.Projection;
+import org.elasticsearch.action.bulk.BulkRetryCoordinatorPool;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.fielddata.IndexFieldDataService;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.Test;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+public class ShardCollectServiceTest extends SQLHttpIntegrationTest {
+
+    @Test
+    public void testReadIsolation() throws Exception {
+        execute("create blob table b1 clustered into 1 shards with (number_of_replicas = 0)");
+        ensureYellow();
+        upload("b1", "foo");
+        upload("b1", "bar");
+        refresh();
+
+        Functions functions = internalCluster().getInstance(Functions.class);
+        CrateDocIndexService crateDocIndexService = mock(CrateDocIndexService.class);
+        when(crateDocIndexService.docInputSymbolVisitor()).thenReturn(new CollectInputSymbolVisitor(functions,
+                                                                      BlobReferenceResolver.INSTANCE));
+        ShardCollectService shardCollectService = new ShardCollectService(
```

Comment:
When I try that I get the following:
http://pastie.org/private/lkr3q0ljlgpty9emxoizfa